You got mail!

SK-Fast opened the pull request:
hai, i add comand n updates n fix n u h stuff!!

+ lookup command
+ guild command
+ stud2brick now had warning
+ uncase sensitive commands
+ URL parsing to ID for lookup/game/user

SK-Fast committed:
URL parsing to id

SK-Fast committed:
LoooOk up comand

SK-Fast committed:
update

+ Colors!!!
+ Guild command!!!!!
+ Lookup stuff idk!!!!!!!!!!!
+ Stud left warnings!!

SK-Fast committed:
Uncase senstive!!!

bluefield-creator commented:
The bot should not take URL arguments at all, makes it a bit pointless having ids. Also bloats up the chat if we were to use it he bot in that scenario, my personal recommendation is to keep it as id only.

I'll write a full analysis later

bluefield-creator commented:
Also commands can be done not case sensitive by modifying no the command string on index.ts, exports.ts should and is required to use Pascal naming for the variables.

bluefield-creator commented:
Change "stud warning" to just informational alert, it's not really anything "risky " and the conversion is fictional

bluefield-creator commented:
Some of the command functions you sent don't have a returned type of Promise<Message<Boolean>> if i remember the type

bluefield-creator commented:
URL parsing parses anything with numbers.

Not just "id" query, try using some of the built-in libraries for parsing.

![image](https://user-images.githubusercontent.com/48862659/147044918-68575dae-e19e-4256-a59b-8975618e186e.jpeg)

bluefield-creator commented:
And make the url parser as a function on utils.

SK-Fast commented:
kkkkkkkkkkkkkkkkkkkkkkkk

SK-Fast committed:
Revert "Uncase senstive!!!"

This reverts commit 5e1915423f910c9d155d4ad148e0728323265aa8.

SK-Fast committed:
Url stuff!!!

+ URL Parse
+ GetNumbersFromURL

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -15,7 +15,51 @@ class stringUtils {
+     * @param string URL that want to be parsed
+     * @returns array Numbers that has been found in URL
+    */
+    public static async GetNumbersFromURL(string: string) {
```
Improper naming, use camel case!

bluefield-creator replied:
Rename argument "string" to "url"

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -15,7 +15,51 @@ class stringUtils {
+        return {
+            URLData: url,
+            searchParams: searchParams_url,
+            spitted: spitted_url
```
splitted*

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -15,7 +15,51 @@ class stringUtils {
+
+    /**
+     * ### Get Numbers From URL
+     * @param string URL that want to be parsed
```
Use proper JSDoc formatting.
* @param `{ typeHere }` `nameOfTheVariable` `Description of the variable`
```js
/**
* @param {string} url URL to be inserted.
*

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -15,7 +15,51 @@ class stringUtils {
+     * @returns array Numbers that has been found in URL
+    */
+    public static async GetNumbersFromURL(string: string) {
+        let result = await stringUtils.URLParse(string)
```
Use const here.

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -15,7 +15,51 @@ class stringUtils {
+        let result = await stringUtils.URLParse(string)
+        let splitted = result.spitted
+
+        let returnresult:Array<any> = []
```
camel case and Type should be:

```js
const returnResult : number[] = [];

```

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -15,7 +15,51 @@ class stringUtils {
+
+        let returnresult:Array<any> = []
+
+        splitted.forEach(function (item:string) {
```
Use ES6 Syntax, this is typescript not JavaScript.

```js
splitted.forEach(item => {})
```

`number` type is inherited from the array, meaning it should always be a number.

bluefield-creator replied:
Thus no need for types on `item` which is not a string?

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -15,7 +15,51 @@ class stringUtils {
+     * }
+    */
+
+    public static async URLParse(string: string) {
```
Change `string` argument to `url`

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -15,7 +15,51 @@ class stringUtils {
+
+    public static async URLParse(string: string) {
+        let url = new URL(string);
+        let searchParams_url = url.searchParams
```
Dont mix snake case with camel case with pascal case... use camelCase only.

searchParamsUrl

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -15,7 +15,51 @@ class stringUtils {
+    public static async URLParse(string: string) {
+        let url = new URL(string);
+        let searchParams_url = url.searchParams
+        let spitted_url:Array<any> = string.split("/")
```
Should be a `const` and the type should be `string[]` with a space after the variable.

Variable name must be camelCase. Not snake case or pascal case.
```ts
const splittedUrl : string[] = url.split("/")
```

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -15,7 +15,51 @@ class stringUtils {
+    /**
+     * ### URL Parse
+     * @param string URL that want to be parsed
+     * @returns Dictionary
```
"Dictionary" does not exist in JavaScript, in javascript they are objects.
Use proper JSDoc formatting.

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -15,7 +15,51 @@ class stringUtils {
+     * ### URL Parse
+     * @param string URL that want to be parsed
+     * @returns Dictionary
+     * {
```
Use proper example, this is not what it returns.

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -15,7 +15,51 @@ class stringUtils {
+
+        splitted.forEach(function (item:string) {
+            if (isNaN(parseInt(item)) == false) {
+                returnresult.push(item)
```
camelCase

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -15,7 +15,51 @@ class stringUtils {
+            }
+        })
+
+        return returnresult
```
camelCase

bluefield-creator reviewed (changes requested):
Fix all of these before merguing. Dont work on anything else.

bluefield-creator commented on package.json:
```diff
@@ -9,7 +9,8 @@
 		"build": "npx tsc",
-		"start": "node dist/index.js"
+		"start": "node dist/index.js",
+		"buildstart": "npx tsc && node dist/index.js"
```
Remove this, build process is not buildstart.

bluefield-creator commented on src/commands/guild.ts:
```diff
@@ -0,0 +1,44 @@
+import fetch from 'node-fetch';
+import { Message, MessageEmbed, MessageSelectMenu } from 'discord.js'
+import { userUtils } from '../utils/userUtils.js'
+export async function guild(message: Message, _arguments: string[]) {
```
No return type.

bluefield-creator commented on src/exports.ts:
```diff
@@ -11,7 +12,8 @@ const commands = {
     game:        game,
-    cookie:      cookie
+    cookie:      cookie,
+    guild:      guild
```
Align this

bluefield-creator reviewed (changes requested):
Change these too

bluefield-creator commented on src/commands/user/lookup.ts:
```diff
@@ -1,13 +1,66 @@
+import fetch from 'node-fetch';
+import { Message, MessageEmbed, MessageSelectMenu } from 'discord.js'
+
+var URLPattern = new RegExp('^(https?:\\/\\/)?'+
```
DONT USE VAR

We use typescript for a reason. Thus use `let` and `const`.

Also what the fuck is this regex

bluefield-creator replied:
Also pretty sure there's url builders builtin js.

bluefield-creator replied:
^^

bluefield-creator commented on src/commands/user/lookup.ts:
```diff
@@ -1,13 +1,66 @@
+
+export async function lookUp(message: Message, _arguments: string[]) {
+
+	let apiURL = `https://api.polytoria.com/v1/games/info?id`;
```
_arguments[1] should be:
```
username
```
or
```
id
```
Use switch cases to select the API to use, do not use URL.

bluefield-creator commented on src/commands/user/lookup.ts:
```diff
@@ -1,13 +1,66 @@
+    '(\\?[;&a-z\\d%_.~+=-]*)?'+
+    '(\\#[-a-z\\d_]*)?$','i');
+
+export async function lookUp(message: Message, _arguments: string[]) {
```
Return type.

skooids commented:
Get rekt noob

Kqpa commented:
Dam

Alyxsqrd commented:
RIP BOZO 😱

StarManTheGamer commented:
ELIMINATED

SK-Fast commented:
korbo

SK-Fast committed:
b

+ add export 2 stringUtils
+ change url checking method
+ add IsURL function in stringUtils

SK-Fast committed:
update lokup

+ shagy make me remove url
+ shagy make me switch case
+ stuff

bluefield-creator commented on src/commands/user/lookup.ts:
```diff
@@ -1,20 +1,20 @@
-		apiURL = "https://api.polytoria.com/v1/users/user?id=" + _arguments[0]
+	switch (_arguments[1]) {
+		case "user":
+			apiURL = `https://api.polytoria.com/v1/users/getbyusername?username=${_arguments[0]}`;
```
Formatting, why is there a space here?

bluefield-creator replied:
Also why are you doing
```js
let apiURL = ...
```
if its not actually being used at all?

SK-Fast committed:
You got mail! Episode 2

+ shaggy make me remove apiUrl first url cuz it's useless
+ add apiErrorHandler

SK-Fast committed:
Merge branch 'main' into main

bluefield-creator commented on package.json:
```diff
@@ -15,13 +15,6 @@
 		"dotenv": "^10.0.0",
 		"node-fetch": "^3.1.0"
 	},
-	"scripts": {
```
Did you just remove scripts and types? Why?

bluefield-creator commented on src/commands/guild.ts:
```diff
@@ -1,12 +1,17 @@
 import { userUtils } from '../utils/userUtils.js'
+import { apiErrorHandler } from '../utils/apiErrorHandler.js'
+
 export async function guild(message: Message, _arguments: string[]) {
```
Still no return type, change apiErrorHandler to responseHandler

bluefield-creator commented on src/commands/guild.ts:
```diff
@@ -1,12 +1,17 @@
 
 	const response = await fetch(apiURL);
 	const data: any = await response.json();
+	const errresult = apiErrorHandler.CheckError(response,data)
```
Use camel case.

bluefield-creator commented on src/commands/guild.ts:
```diff
@@ -1,12 +1,17 @@
 	const response = await fetch(apiURL);
 	const data: any = await response.json();
+	const errresult = apiErrorHandler.CheckError(response,data)
+	if (errresult.HasError == true) return message.channel.send(`${errresult.DisplayText}`)
```
CAMEL CASE

bluefield-creator commented on src/commands/user/lookup.ts:
```diff
@@ -19,7 +20,10 @@ export async function lookUp(message: Message, _arguments: string[]) {
 	const response = await fetch(apiURL);
 	const data: any = await response.json();
-	if (data.Success !== true) return message.channel.send('There was an unexpected error.')
+	const errresult = apiErrorHandler.CheckError(response,data)
```
Camel case...

bluefield-creator commented on src/exports.ts:
```diff
@@ -1,5 +1,5 @@
-import * as userCommands from './commands/user.js'
-import {ping} from './commands/ping.js'
+import * as userCommands from './commands/user.js';
+import { ping } from './commands/ping.js';
```
Why did you add this? It's against code style guidelines, check eslint.

bluefield-creator commented on src/utils/apiErrorHandler.ts:
```diff
@@ -0,0 +1,49 @@
+let DisplayTexts: any = {
```
Should be a const, and under an Error Type.

bluefield-creator commented on src/utils/apiErrorHandler.ts:
```diff
@@ -0,0 +1,49 @@
+            ActualError: "Actual Error from server"
+        }
+     */
+    public static CheckError(response: any,data: any) {
```
Camel case

bluefield-creator commented on src/utils/apiErrorHandler.ts:
```diff
@@ -0,0 +1,49 @@
+     * @param response Gets request type from node-fetch
+     * @param data Gets request.json from node-fetch
+     * @returns Dictionary
+     * {
```
camel case

bluefield-creator commented on src/utils/apiErrorHandler.ts:
```diff
@@ -0,0 +1,49 @@
+     * Check Error
+     * @param response Gets request type from node-fetch
+     * @param data Gets request.json from node-fetch
+     * @returns Dictionary
```
Dictionary does not exist in javascript, use proper naming.

bluefield-creator replied:
Most of your mistakes were notified in the past 4 reviews, you still commit them. We have eslint, and prettier for a reason, use the tools.

bluefield-creator commented on src/utils/apiErrorHandler.ts:
```diff
@@ -0,0 +1,49 @@
+     */
+    public static CheckError(response: any,data: any) {
+
+        let result: any = {
```
Add result type, should be a const too.

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -1,83 +1,83 @@
+    /**
+     * ### URL Parse
+     * @param string URL that want to be parsed
+     * @returns Dictionary
```
Dictionary does not exist in javascript.

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -1,83 +1,83 @@
+     * @param string URL that want to be parsed
+     * @returns Dictionary
+     * {
+     *  URLData: origin URL result,
```
Use proper example parameter.

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -1,83 +1,83 @@
+        let spitted_url:Array<any> = string.split("/")
+
+        return {
+            URLData: url,
```
Camel case.

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -1,83 +1,83 @@
+     * @param string URL that want to be parsed
+     * @returns array Numbers that has been found in URL
+    */
+    public static GetNumbersFromURL(string: string) {
```
Camel case.

bluefield-creator replied:
And I told you, string had to be named to url for less confusions.

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -1,83 +1,83 @@
+            }
+        })
+
+        return returnresult
```
Camel case.

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -1,83 +1,83 @@
+
+        let returnresult:Array<any> = []
+
+        splitted.forEach(function (item:string) {
```
Space on types.

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -1,83 +1,83 @@
+        let result = stringUtils.URLParse(string)
+        let splitted = result.spitted
+
+        let returnresult:Array<any> = []
```
Use `any[]` not `Array<any>` 

Also camel case again.

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -1,83 +1,83 @@
+     * @param string string that you want to checks that's url or not
+     * @returns bool returns if it were url or not
+     */
+    public static IsURL(string: string) {
```
Camel case again.

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -1,83 +1,83 @@
+
+        // Try stuff, If URL fail to parse, that means it's not vaild URL.
+        try {
+            test_url = new URL(string)
```
camel case.

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -1,83 +1,83 @@
+     * @returns bool returns if it were url or not
+     */
+    public static IsURL(string: string) {
+        let test_url;
```
inappropiate naming

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -1,83 +1,83 @@
+        try {
+            test_url = new URL(string)
+        } catch(_) {
+            return false
```
Incorrect use of try and catch functionality.

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -1,83 +1,83 @@
+            return false
+        }
+
+        return test_url.protocol === "http:" || test_url.protocol === "https:"; // Allow only site protocols. filter out something like javascript:void(0)
```
Use this properly please...

SK-Fast commented:
wooo camel, ok

bluefield-creator commented:
> wooo camel, ok

Do you know what camel case is?

SK-Fast committed:
caramel case

SK-Fast committed:
c

+ caramel case | the sequel
+ check url use regular expressions instead of random try catch

SK-Fast commented:
> > wooo camel, ok
> 
> Do you know what camel case is?

isCaramelCase

SK-Fast committed:
isurl the rename

SK-Fast committed:
caramel case episode 3

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -75,7 +75,7 @@ export class stringUtils {
      * @returns bool returns if it were url or not
      */
-    public static StrisURL(string: string) {
+    public static strIsURL(string: string) {
```
Rename just to isURL, and once more and again... parameter name has to be url...

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -75,7 +75,7 @@ export class stringUtils {
      */
-    public static StrisURL(string: string) {
+    public static strIsURL(string: string) {
         return !!url_pattern.test(string);
```
Camel case means ALL of the code has to be camel case.

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -1,3 +1,10 @@
+var url_pattern = new RegExp('^(https?:\\/\\/)?'+
```
Why did you bring this back? also `var` is not typescript.

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -30,7 +37,7 @@ export class stringUtils {
     */
 
-    public static URLParse(string: string) {
+    public static urlParse(string: string) {
```
You missed it again?? CHANGE `string` to `url` my god. Parameter names need to be concise.

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -30,7 +37,7 @@ export class stringUtils {
-    public static URLParse(string: string) {
+    public static urlParse(string: string) {
         let url = new URL(string);
         let searchParams_url = url.searchParams
```
What the fuck? 

DONT MIX camelCase with snake_case!

Don't use Array<any> use any[];

Add a space between types!

SK-Fast committed:
damn stinky caraml case

SK-Fast committed:
s p a c e  r e m o v a l

SK-Fast committed:
const cons

SK-Fast committed:
cosnt const

bluefield-creator commented on src/commands/currency/studToBrick.ts:
```diff
@@ -3,16 +3,22 @@ import {Message, MessageEmbed} from 'discord.js'
+    studs /= 15
+    studs = Math.round(studs)
+    let studremoved = Number(_arguments[0]) % 15
+    let studshould = studs - studremoved
```
camel case

bluefield-creator commented on src/commands/game.ts:
```diff
@@ -1,11 +1,18 @@
-	const data: any = await response.json()
+	const response = await fetch(apiURL);
+	const data: any = await response.json();
+	const errresult = responseHandler.checkError(response,data)
```
camel case

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -54,7 +54,7 @@ export class stringUtils {
      * @returns array Numbers that has been found in URL
     */
-    public static GetNumbersFromURL(string: string) {
+    public static getNumbersFromURL(string: string) {
```
change `string` parameter to `url`

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -54,7 +54,7 @@ export class stringUtils {
     */
-    public static GetNumbersFromURL(string: string) {
+    public static getNumbersFromURL(string: string) {
         let result = stringUtils.urlParse(string)
```
Should be a const.

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -54,7 +54,7 @@ export class stringUtils {
-    public static GetNumbersFromURL(string: string) {
+    public static getNumbersFromURL(string: string) {
         let result = stringUtils.urlParse(string)
         let splitted = result.spitted
```
should be a const, and array type.

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -1,4 +1,4 @@
-var url_pattern = new RegExp('^(https?:\\/\\/)?'+
+const url_pattern = new RegExp('^(https?:\\/\\/)?'+
```
Why is this here? It's unused in favor of builtin javascript URl parsing.

bluefield-creator commented on src/utils/responseHandler.ts:
```diff
@@ -18,7 +18,7 @@ export class responseHandler {
     public static checkError(response: any,data: any) {
 
-        let result: any = {
+        const result: any = {
```
infer `result` type. If you need it, write it down. Should be relatively easy.

```ts

type result {
   value: dataType
}
```

bluefield-creator commented on src/utils/responseHandler.ts:
```diff
@@ -18,7 +18,7 @@ export class responseHandler {
+        const result: any = {
             HasError: false,
             StatusCode: 0,
             DisplayText: "Unknown Error.",
```
camel case...

SK-Fast committed:
god damn fixes

+ Dictionary built different
+ types br br br
+ god damn s p a c e  after colon

bluefield-creator commented on src/commands/game.ts:
```diff
@@ -1,11 +1,18 @@
+	const response = await fetch(apiURL);
+	const data: any = await response.json();
+	const errresult = responseHandler.checkError(response,data)
+	if (errresult.HasError == true) return message.channel.send(`${errresult.DisplayText}`)
```
camelcase

bluefield-creator commented on src/commands/user/lookup.ts:
```diff
@@ -1,7 +1,72 @@
+
+export async function lookUp(message: Message, _arguments: string[]) {
+
+	let apiURL;
```
Infer type `IapiEndpoints`

SK-Fast replied:
wdym `IapiEndpoints`  ?     ? ?

bluefield-creator replied:
Create a type that holds API endpoints...

```ts

type IapiEndpoints = "url1" | "url2" | "url3"...
```

SK-Fast replied:
> Create a type that holds API endpoints...
> 
> ```ts
> type IapiEndpoints = "url1" | "url2" | "url3"...
> ```

there is id behind them
![image](https://user-images.githubusercontent.com/60780448/147252557-4a1ac1ca-9ce1-4b4c-9a7b-7cd6abdef143.png)

![image](https://user-images.githubusercontent.com/60780448/147252504-a6666c8d-7975-4b02-b76e-2333d34ff864.png)

bluefield-creator commented on src/utils/responseHandler.ts:
```diff
@@ -0,0 +1,51 @@
+            ActualError: string
+        }
+     */
+    public static checkError(response: Response,data: any) {
```
space between comas

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -1,21 +1,83 @@
+    public static urlParse(url: string) {
+        let urlConstruct = new URL(url);
+        let searchParams_url = urlConstruct.searchParams
+        let spitted_url: Array<string> = url.split("/")
```
Use type `string[]` not `Array<string>`

bluefield-creator commented:
I'm reviewing only one more batch of updates. No more. Make sure you fix everything, otherwise not getting mergued.

Alyxsqrd committed:
Yeah

Alyxsqrd committed:
Merge branch 'main' into pr/1

Alyxsqrd committed:
No npm

Alyxsqrd committed:
Migrate from node-fetch to axios

bluefield-creator commented:
I personally disapprove the fact that alyx had to fix your problems @SK-Fast, consdier this a bad grade.

bluefield-creator commented on src/commands/guild.ts:
```diff
@@ -1,49 +1,59 @@
 		description: data.Description,
 		thumbnail: {
-			url: `${data.Thumbnail}`
+			url: data.thumbnail
```
This does not work, in the response data thumbnail comes as `Thumbnail` and it's not considered a string for the Embed type for some reason.

Alyxsqrd committed:
Cleanup string utils

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -1,32 +1,31 @@
-    '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*'+
-    '(\\?[;&a-z\\d%_.~+=-]*)?'+ 
-    '(\\#[-a-z\\d_]*)?$','i');
+const url_pattern = new RegExp(
```
Why is this still here??

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -37,47 +36,46 @@ export class stringUtils {
+	public static urlParse(url: string) {
+		let urlConstruct = new URL(url)
+		let searchParams_url = urlConstruct.searchParams
+		let spitted_url: Array<string> = url.split('/')
```
Wrong type again.

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -37,47 +36,46 @@ export class stringUtils {
+		let returnResult: {URLData: URL; searchParams: URLSearchParams; spitted: Array<string>} = {
+			URLData: urlConstruct,
+			searchParams: searchParams_url,
+			spitted: spitted_url
```
Camel case...

Troplo commented:
How many comments can someone make

Alyxsqrd committed:
Phicksed typo

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -1,81 +1,36 @@
-		let splitted = result.spitted
+	public static getNumbersFromURL(string: string): number[] {
+		const parsed = stringUtils.urlParse(string)
+		const result: number[] = []
```
Should be a `let` since we push items.

Alyxsqrd committed:
Cleanup a bit

bluefield-creator commented on src/utils/stringUtils.ts:
```diff
@@ -16,8 +16,8 @@ export class stringUtils {
+	 * @param {string} URL that want to be parsed
+	 * @returns {number[]} Numbers that has been found in URL
 	 */
 	public static getNumbersFromURL(string: string): number[] {
```
url not string

Kqpa commented:
my guy gave up

SK-Fast committed:
You got mail Ep 12 | Last episode?

+ camel case
+ space
+ s p a c e
+ s  p  a  c  e
+ .eslintrc
+ create type IapiEndpoints

bluefield-creator reviewed (approved):
Passes testing, going to review a bit more then merguing.

bluefield-creator commented:
> Passes testing, going to review a bit more then merguing.

asides recommendation of using more Axios types (which is the reason @Alyxsqrd wanted to use axios to begin with) everything seems ok, I'll pass it on build now, and if it works merguing. 

Will begin setting up CI for this.

bluefield-creator commented:
There's a small mistake at Eslintrc json, ecmaversion cannot be 13, for latest ecmascript use keyword "latest"

```json
{
    "env": {
        "es2021": true,
        "node": true
   },
    "extends": [
        "standard"
    ],
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaVersion": "latest",
        "sourceType": "module"
   },
    "plugins": [
        "@typescript-eslint"
    ],
    "rules": {
   }
}

```

bluefield-creator commented:
~~Another thing I noticed, is that someone Imported discord.js to the index.d.ts file, remove the import.~~
NVM it is being used for Icommand type.

bluefield-creator commented:
![image](https://user-images.githubusercontent.com/48862659/147358231-87c72323-615c-4183-9238-46cbf862f350.png)
Noticed that the eslint configuration is slightly off.

bluefield-creator commented:
https://user-images.githubusercontent.com/48862659/147358352-e50cfc23-df92-4e9c-bbab-db4bc465917e.mp4

eslint configuration goes against prettier configuration.

SK-Fast committed:
prettier moment
